pg_search/src/postgres/mod.rs

```diff
@@ -622,6 +647,33 @@ impl ParallelScanState {
         segments
     }
 
+    /// Wait until the leader has initialized the parallel state.
+    /// Workers call this before reading segment data.
+    fn wait_for_initialization(&mut self) {
+        let deadline =
+            std::time::Instant::now() + std::time::Duration::from_secs(PARALLEL_INIT_TIMEOUT_SECS);
+
+        loop {
+            {
+                let _mutex = self.acquire_mutex();
+                if self.nsegments != PARALLEL_STATE_UNINITIALIZED {
+                    // Leader has initialized the state
+                    return;
+                }
+            }
+
+            // Check for interrupts to allow query cancellation
+            pgrx::check_for_interrupts!();
+
+            if std::time::Instant::now() >= deadline {
+                pgrx::error!("Timeout waiting for parallel scan leader to initialize segment data");
+            }
+
+            // Brief sleep to avoid busy-waiting
+            std::thread::sleep(std::time::Duration::from_micros(PARALLEL_INIT_SLEEP_MICROS));
```

Comment:
Please refer to https://github.com/paradedb/paradedb/issues/3489 here in a TODO: ideally we can remove all of these busy waits at some point.

Reply:
Actually, assuming it has gone in, please switch to using https://github.com/paradedb/paradedb/pull/3800